### DISCUSSION TITLE
Error when launching TShock 4.5.17 on Ubuntu 20.04

### DISCUSSION THREAD

**Discussion by Aresiel** [Pryaxis/TShock#2620] [Q&A]
Hello,

I'm trying to run [TShock 4.5.17](https://github.com/Pryaxis/TShock/releases/tag/v4.5.17) on my server running ubuntu 20.04 but am getting the following error when trying to start TerrariaServer.exe with mono.
`Starting process: /usr/bin/mono ./server/TerrariaServer.exe -ip 0.0.0.0   -announcementboxrange -1 -config ./serverconfig.txt﻿﻿﻿﻿Error Logging Enabled.
=]0;Terraria Server v1.4.3.6TerrariaAPI Version: 2.1.0.0 (Protocol v1.4.3.6 (248), OTAPI 1.4.3.6)
Server crash: 4/24/2022 4:54:27 PM
System.NullReferenceException: Object reference not set to an instance of an object
  at Terraria.Main.DedServ () [0x00a9e] in <dff31e9b1ce9459785293368a701a09d>:0 
  at Terraria.Program.LaunchGame (System.String[] args, System.Boolean monoArgs) [0x000ac] in <dff31e9b1ce9459785293368a701a09d>:0 

Please send crashlog.txt to support@terraria.org`

I would greatly appreciate any help regarding this.

Update: This only seems to occur when starting the server with the world I already have (that works on other terraria servers etc ofc), if I instead let it autocreate a server, it does start however there are no traces of TShock, just looks like a vanilla server. If I then turn off the server and start it again, it works, however, there are yet again no traces of TShock.

**Comment by Kojirremer** [comment]
I don't really know what's the problem, but just for troubleshooting purposes:

1. Are you sure you are using the latest mono version from their website (not your default repo)?
2. If "yes" to previous question, have you tried running TShock without any launch commandline parameters?

As for mono, quoting @Ristovski on Discord:
"_Btw, the most minimal mono setup for TShock is: **mono-runtime** and the following packages: **libmono-system-drawing4.0-cil libmono-system-net-http4.0-cil libmono-system-data4.0-cil libmono-system-transactions4.0-cil libmono-system-runtime-serialization4.0-cil**. Saves quite a bit of disk space compared to **mono-complete**_"

**Reply by Aresiel** [reply]
> I don't really know what's the problem, but just for troubleshooting purposes:
> 
>     1. Are you sure you are using the latest mono version from their website (not your default repo)?
> 
>     2. If "yes" to previous question, have you tried running TShock without any launch commandline parameters?
> 
> 
> As for mono, quoting @Rivstoski on Discord: "_Btw, the most minimal mono setup for TShock is: **mono-runtime** and the following packages: **libmono-system-drawing4.0-cil libmono-system-net-http4.0-cil libmono-system-data4.0-cil libmono-system-transactions4.0-cil libmono-system-runtime-serialization4.0-cil**. Saves quite a bit of disk space compared to **mono-complete**_"

I've installed mono-complete by following [the guide linked by TShock](https://www.mono-project.com/download/stable/#download-lin). I have not tried launching it without any parameters, my launching command is `mono ./server/TerrariaServer.exe -ip {ip} -config ./serverconfig.txt` for reference.

**Comment by Aresiel** [comment]
Update: It only happens when I try to start the server with the world I already have.

**Comment by Aresiel** [comment]
However, letting it autocreate a world leaves me with a completely vanilla server, with no traces of tshock.

**Comment by Aresiel** [comment]
This remains the case even when starting the server again, with the empty world.
